larq/layers.py

```diff
@@ -76,6 +76,64 @@ def get_config(self):
         return {**super().get_config(), **config}
 
 
+class QuantizerDepthwiseBase(tf.keras.layers.Layer):
```

Comment:
I would really like to find a better way of moving this logic to a shared base class, since we're basically copy pasting the same code for `QuantizerBase`, `SeperableQuantizerBase` and now `QuantizerDepthwiseBase`. I'd be OK with doing this in a future PR, but as of now this will become a large maintenance burden if we start adding more layers.

Reply:
Agreed. I would argue the logic needs to be implemented at a variable level, but this is tricky to do as it touches a lot of tf internal logic.